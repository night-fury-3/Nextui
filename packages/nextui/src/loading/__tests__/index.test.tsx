import React from 'react';
import { mount } from 'enzyme';
import Loading from '../index';

describe('Loading', () => {
  it('should render correctly', () => {
    const wrapper = mount(<Loading />);
    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).toMatchSnapshot();
  });

  it('should work with children', () => {
    const wrapper = mount(<Loading>Loading</Loading>);
    expect(wrapper.find('label').text()).toContain('Loading');
  });

  it('should work with different colors', () => {
    const wrapper = mount(
      <div>
        <Loading color="#f81ce5" />
        <Loading color="primary" />
        <Loading color="success" />
        <Loading color="secondary" />
        <Loading color="warning" />
        <Loading color="error" />
      </div>
    );
    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).toMatchSnapshot();
  });

  it('should work with different text colors', () => {
    const wrapper = mount(
      <div>
        <Loading textColor="#f81ce5" />
        <Loading textColor="primary" />
        <Loading textColor="success" />
        <Loading textColor="secondary" />
        <Loading textColor="warning" />
        <Loading textColor="error" />
      </div>
    );
    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).toMatchSnapshot();
  });

  it('should work with different sizes', () => {
    const wrapper = mount(
      <div>
        <Loading size="mini" />
        <Loading size="small" />
        <Loading size="medium" />
        <Loading size="large" />
        <Loading size="xlarge" />
      </div>
    );
    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).toMatchSnapshot();
  });

  it('should work with different types', () => {
    const wrapper = mount(
      <div>
        <Loading type="default" />
        <Loading type="spinner" />
        <Loading type="points" />
        <Loading type="points-opacity" />
        <Loading type="gradient" />
      </div>
    );
    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).toMatchSnapshot();
  });

  it('should work with text in spinner type', () => {
    const wrapper = mount(<Loading type="spinner">Spinner</Loading>);
    expect(wrapper.find('.spinner').text()).toContain('Spinner');
  });
});
