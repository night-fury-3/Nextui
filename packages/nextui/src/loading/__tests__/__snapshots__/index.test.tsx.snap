// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render correctly 1`] = `"<div class=\\"nextui-c-bOoPrE\\"><span class=\\"nextui-c-chTLhx nextui-c-chTLhx-cRKUxm-color-primary nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-dsbHMJ-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\" aria-label=\\"Loading\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div>"`;

exports[`Loading should render correctly 2`] = `[Function]`;

exports[`Loading should work with different colors 1`] = `"<div><div class=\\"nextui-c-bOoPrE\\"><span class=\\"nextui-c-chTLhx nextui-c-chTLhx-cRKUxm-color-primary nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-dsbHMJ-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\" aria-label=\\"Loading\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE\\"><span class=\\"nextui-c-chTLhx nextui-c-chTLhx-kesFop-color-success nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-dsbHMJ-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\" aria-label=\\"Loading\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE\\"><span class=\\"nextui-c-chTLhx nextui-c-chTLhx-iEzErA-color-secondary nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-dsbHMJ-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\" aria-label=\\"Loading\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE\\"><span class=\\"nextui-c-chTLhx nextui-c-chTLhx-fzAkwg-color-warning nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-dsbHMJ-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\" aria-label=\\"Loading\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE\\"><span class=\\"nextui-c-chTLhx nextui-c-chTLhx-bYEXDy-color-error nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-dsbHMJ-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\" aria-label=\\"Loading\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div></div>"`;

exports[`Loading should work with different colors 2`] = `[Function]`;

exports[`Loading should work with different sizes 1`] = `"<div><div class=\\"nextui-c-bOoPrE\\"><span class=\\"nextui-c-chTLhx nextui-c-chTLhx-cRKUxm-color-primary nextui-c-chTLhx-bboDzR-size-xs nextui-c-chTLhx-dsbHMJ-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\" aria-label=\\"Loading\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE\\"><span class=\\"nextui-c-chTLhx nextui-c-chTLhx-cRKUxm-color-primary nextui-c-chTLhx-eWttbs-size-sm nextui-c-chTLhx-dsbHMJ-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\" aria-label=\\"Loading\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE\\"><span class=\\"nextui-c-chTLhx nextui-c-chTLhx-cRKUxm-color-primary nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-dsbHMJ-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\" aria-label=\\"Loading\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE\\"><span class=\\"nextui-c-chTLhx nextui-c-chTLhx-cRKUxm-color-primary nextui-c-chTLhx-eWTJwO-size-lg nextui-c-chTLhx-dsbHMJ-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\" aria-label=\\"Loading\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE\\"><span class=\\"nextui-c-chTLhx nextui-c-chTLhx-cRKUxm-color-primary nextui-c-chTLhx-fYAaIu-size-xl nextui-c-chTLhx-dsbHMJ-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\" aria-label=\\"Loading\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div></div>"`;

exports[`Loading should work with different sizes 2`] = `[Function]`;

exports[`Loading should work with different text colors 1`] = `"<div><div class=\\"nextui-c-bOoPrE\\"><span class=\\"nextui-c-chTLhx nextui-c-chTLhx-cRKUxm-color-primary nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-dsbHMJ-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\" aria-label=\\"Loading\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE\\"><span class=\\"nextui-c-chTLhx nextui-c-chTLhx-cRKUxm-color-primary nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-dsbHMJ-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\" aria-label=\\"Loading\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE\\"><span class=\\"nextui-c-chTLhx nextui-c-chTLhx-cRKUxm-color-primary nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-dsbHMJ-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\" aria-label=\\"Loading\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE\\"><span class=\\"nextui-c-chTLhx nextui-c-chTLhx-cRKUxm-color-primary nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-dsbHMJ-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\" aria-label=\\"Loading\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE\\"><span class=\\"nextui-c-chTLhx nextui-c-chTLhx-cRKUxm-color-primary nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-dsbHMJ-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\" aria-label=\\"Loading\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div></div>"`;

exports[`Loading should work with different text colors 2`] = `[Function]`;

exports[`Loading should work with different types 1`] = `"<div><div class=\\"nextui-c-bOoPrE\\"><span class=\\"nextui-c-chTLhx nextui-c-chTLhx-cRKUxm-color-primary nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-dsbHMJ-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\" aria-label=\\"Loading\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-fpphdg nextui-c-fpphdg-fPWnWX-size-md nextui-c-fpphdg-kNkCaB-color-primary nextui-spinner undefined\\"><div class=\\"nextui-c-jTKaZN nextui-spinner-container\\" aria-label=\\"Loading\\"><span class=\\"nextui-c-gsMpnF\\"></span><span class=\\"nextui-c-gsMpnF\\"></span><span class=\\"nextui-c-gsMpnF\\"></span><span class=\\"nextui-c-gsMpnF\\"></span><span class=\\"nextui-c-gsMpnF\\"></span><span class=\\"nextui-c-gsMpnF\\"></span><span class=\\"nextui-c-gsMpnF\\"></span><span class=\\"nextui-c-gsMpnF\\"></span><span class=\\"nextui-c-gsMpnF\\"></span><span class=\\"nextui-c-gsMpnF\\"></span><span class=\\"nextui-c-gsMpnF\\"></span><span class=\\"nextui-c-gsMpnF\\"></span></div></div><div class=\\"nextui-c-bOoPrE\\"><span class=\\"nextui-c-chTLhx nextui-c-chTLhx-cRKUxm-color-primary nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-coWejU-type-points nextui-c-chTLhx-dhbJvz-cv nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-points\\" aria-label=\\"Loading\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE\\"><span class=\\"nextui-c-chTLhx nextui-c-chTLhx-cRKUxm-color-primary nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-hQQprQ-type-points-opacity nextui-c-chTLhx-dhbJvz-cv nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-points-opacity\\" aria-label=\\"Loading\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE\\"><span class=\\"nextui-c-chTLhx nextui-c-chTLhx-cRKUxm-color-primary nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-cQPBpi-type-gradient nextui-c-chTLhx-icvvata-css nextui-loading nextui-loading-gradient\\" aria-label=\\"Loading\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div></div>"`;

exports[`Loading should work with different types 2`] = `[Function]`;
